apps/zotonic_mod_cron/src/mod_cron.erl

```diff
@@ -88,26 +96,67 @@ handle_info(_Info, State) ->
 code_change(_Vsn, State, _Extra) ->
     {ok, State}.
 
-terminate(_Reason, #state{ tick_job_refs = TickJobs }) ->
-    lists:foreach(fun(JobRef) -> _ = erlcron:cancel(JobRef) end, TickJobs),
+terminate(_Reason, #state{ context = Context, jobs = Jobs}) ->
+    z_notifier:detach(module_activate, self(), Context),
+    z_notifier:detach(module_deactivate, self(), Context),
+    maps:foreach(fun(_Module, ModuleJobs) -> _ = cancel_jobs(ModuleJobs) end, Jobs),
```

Comment:
Do the jobs also get killed if the module process crashes?

And what happens if a job crashes?

Reply:
- When the module crashes the module manager will do a module_deactivate notification. This will cancel all its jobs in mod_cron.
- Each job is a process supervised by erlcron. When it crashes a message will be logged and the job will be restarted.

I added extra error handling for when the module has made a wrong job clause. For example when a non existing function is configured. If that happens a message will be logged.

Reply:
And if `mod_cron` crashes/stops?  What does the cron app do?